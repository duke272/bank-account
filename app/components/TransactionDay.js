import React from 'react';
import PropTypes from 'prop-types';
import { List, ListItem } from '@material-ui/core';
import EventIcon from '@material-ui/icons/Event';
import Transaction from './Transaction';

const TransactionDay = ({ transaction }) => {
  // @TODO do the date in selector, not here
  const date = new Date(transaction.date);

  return (
    <>
      <ListItem
        style={{
          color: '#999',
          padding: '10px',
          backgroundColor: '#eee',
        }}
      >
        <EventIcon />
        {date.toLocaleDateString('nl-NL', {
          day: '2-digit',
          month: '2-digit',
          year: 'numeric',
        })}
      </ListItem>
      <ListItem style={{ padding: '0' }}>
        <List style={{ padding: '0 0 0 10px', color: '#000', width: '100%' }}>
          {transaction.transactions.map((trans, transactionKey) => (
            <Transaction
              trans={trans}
              // eslint-disable-next-line react/no-array-index-key
              key={`${transaction.id}-${transactionKey}`}
            />
          ))}
        </List>
      </ListItem>
    </>
  );
};

TransactionDay.propTypes = {
  transaction: PropTypes.shape({
    date: PropTypes.string.isRequired,
    transactions: PropTypes.array.isRequired,
  }),
};

export default TransactionDay;
