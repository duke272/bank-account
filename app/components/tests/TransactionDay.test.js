import React from 'react';
import renderer from 'react-test-renderer';
import TransactionDay from '../TransactionDay';

const transaction = {
  id: '0315778504.2017-07-19',
  iban: 'NL92RABO0315778504',
  date: '2017-07-19T20:40:29.000Z',
  balances: [
    { amount: 4861.79, debit_credit: 'credit', date: '2017-07-18' },
    { amount: 4836.77, debit_credit: 'credit', date: '2017-07-19' },
  ],
  transactions: [
    {
      iban: '',
      name: 'Bizcuit',
      amount: 0.02,
      debit_credit: 'debit',
      date: '2017-07-19',
      description: 'Betalingsnummer 2913045c',
    },
    {
      iban: '',
      name: 'Bizcuit',
      amount: 10,
      debit_credit: 'debit',
      date: '2017-07-19',
      description: 'Betalingsnummer e1e33717',
    },
    {
      iban: '',
      name: 'Bizcuit',
      amount: 15,
      debit_credit: 'debit',
      date: '2017-07-19',
      description: 'Betalingsnummer 67805d60',
    },
  ],
};

describe('<TransactionDay />', () => {
  it('should render', () => {
    const tree = renderer
      .create(<TransactionDay transaction={transaction} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
