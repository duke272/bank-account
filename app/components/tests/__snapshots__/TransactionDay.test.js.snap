// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionDay /> should render 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
    style={
      Object {
        "backgroundColor": "#eee",
        "color": "#999",
        "padding": "10px",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
      />
    </svg>
    19-07-2017
  </li>,
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
    style={
      Object {
        "padding": "0",
      }
    }
  >
    <ul
      className="MuiList-root MuiList-padding"
      style={
        Object {
          "color": "#000",
          "padding": "0 0 0 10px",
          "width": "100%",
        }
      }
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57z"
              />
            </svg>
            +0,02
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <p
              title="omschrijving"
            >
              Betalingsnummer 2913045c
            </p>
            <p
              title="naam"
            >
              Bizcuit
            </p>
            
          </p>
        </div>
      </li>
      <hr
        className="MuiDivider-root"
      />
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57z"
              />
            </svg>
            +10,00
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <p
              title="omschrijving"
            >
              Betalingsnummer e1e33717
            </p>
            <p
              title="naam"
            >
              Bizcuit
            </p>
            
          </p>
        </div>
      </li>
      <hr
        className="MuiDivider-root"
      />
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57z"
              />
            </svg>
            +15,00
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <p
              title="omschrijving"
            >
              Betalingsnummer 67805d60
            </p>
            <p
              title="naam"
            >
              Bizcuit
            </p>
            
          </p>
        </div>
      </li>
      <hr
        className="MuiDivider-root"
      />
    </ul>
  </li>,
]
`;
